### ISSUE TITLE
How to know whether NNI is using GPU?

### ISSUE THREAD

**Issue by RuojiWang** [microsoft/nni#4484]
**Describe the issue**:
i can run the example code of minst_pytorch. i guess i set NNI to use GPU, i don't know that for sure. 
whatever you set in the config.yml, you can always see 'no_cuda': False in the trial.log(i put trial.log under this)

**Environment**:
- NNI version: 2.6
- Training service (local|remote|pai|aml|etc): local
- Client OS: ubuntu 18.04
- Server OS (for remote mode only):
- Python version: 3.8
- PyTorch/TensorFlow version: 1.10.1
- Is conda/virtualenv/venv used?:  virtualenv in anaconda
- Is running in Docker?: no

**Configuration**:
 - Experiment config (remember to remove secrets!):
 
 searchSpaceFile: search_space.json
trialCommand: python3 mnist.py  # NOTE: change "python3" to "python" if you are using Windows

trialGpuNumber: 1
trialConcurrency: 1
maxTrialNumber: 10
maxExperimentDuration: 1h

tuner:
  name: TPE
  classArgs:
    optimize_mode: maximize
trainingService:
  platform: local

trainingService:
  platform: local
  useActiveGpu: true           # NOTE: Use "true" if you are using an OS with graphical interface (e.g. Windows 10, Ubuntu desktop)
  maxTrialNumberPerGpu: 1      #   Reason and details:  https://nni.readthedocs.io/en/latest/reference/experiment_config.html#useactivegpu
 
 - Search space:
{
    "batch_size": {"_type":"choice", "_value": [16, 32, 64, 128]},
    "hidden_size":{"_type":"choice","_value":[128, 256, 512, 1024]},
    "lr":{"_type":"choice","_value":[0.0001, 0.001, 0.01, 0.1]},
    "momentum":{"_type":"uniform","_value":[0, 1]}
}

**Log message**:
 - nnimanager.log:
 [2022-01-20 04:43:25] INFO (NNIDataStore) Datastore initialization done
[2022-01-20 04:43:25] INFO (RestServer) RestServer start
[2022-01-20 04:43:25] WARNING (NNITensorboardManager) Tensorboard may not installed, if you want to use tensorboard, please check if tensorboard installed.
[2022-01-20 04:43:25] INFO (RestServer) RestServer base port is 8080
[2022-01-20 04:43:25] INFO (main) Rest server listening on: http://0.0.0.0:8080
[2022-01-20 04:43:26] INFO (NNIManager) Starting experiment: 0zv4dxuc
[2022-01-20 04:43:26] INFO (NNIManager) Setup training service...
[2022-01-20 04:43:26] INFO (LocalTrainingService) Construct local machine training service.
[2022-01-20 04:43:26] INFO (NNIManager) Setup tuner...
[2022-01-20 04:43:26] INFO (NNIManager) Change NNIManager status from: INITIALIZED to: RUNNING
[2022-01-20 04:43:26] INFO (NNIManager) Add event listeners
[2022-01-20 04:43:26] INFO (LocalTrainingService) Run local machine training service.
[2022-01-20 04:43:26] WARNING (GPUScheduler) gpu_metrics file does not exist!
[2022-01-20 04:43:27] INFO (NNIManager) NNIManager received command from dispatcher: ID, 
[2022-01-20 04:43:27] INFO (NNIManager) NNIManager received command from dispatcher: TR, {"parameter_id": 0, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 512, "lr": 0.0001, "momentum": 0.7760213206833312}, "parameter_index": 0}
[2022-01-20 04:43:31] INFO (NNIManager) submitTrialJob: form: {
  sequenceId: 0,
  hyperParameters: {
    value: '{"parameter_id": 0, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 512, "lr": 0.0001, "momentum": 0.7760213206833312}, "parameter_index": 0}',
    index: 0
  },
  placementConstraint: { type: 'None', gpus: [] }
}
[2022-01-20 04:43:37] INFO (NNIManager) Trial job Bqb0p status changed from WAITING to RUNNING
[2022-01-20 04:45:42] INFO (NNIManager) Trial job Bqb0p status changed from RUNNING to SUCCEEDED
[2022-01-20 04:45:42] INFO (NNIManager) NNIManager received command from dispatcher: TR, {"parameter_id": 1, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 512, "lr": 0.001, "momentum": 0.7308649721316198}, "parameter_index": 0}
[2022-01-20 04:45:47] INFO (NNIManager) submitTrialJob: form: {
  sequenceId: 1,
  hyperParameters: {
    value: '{"parameter_id": 1, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 512, "lr": 0.001, "momentum": 0.7308649721316198}, "parameter_index": 0}',
    index: 0
  },
  placementConstraint: { type: 'None', gpus: [] }
}
[2022-01-20 04:45:53] INFO (NNIManager) Trial job iTtJE status changed from WAITING to RUNNING
[2022-01-20 04:47:53] INFO (NNIManager) Trial job iTtJE status changed from RUNNING to SUCCEEDED
[2022-01-20 04:47:53] INFO (NNIManager) NNIManager received command from dispatcher: TR, {"parameter_id": 2, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 128, "lr": 0.0001, "momentum": 0.4727135784273845}, "parameter_index": 0}
[2022-01-20 04:47:58] INFO (NNIManager) submitTrialJob: form: {
  sequenceId: 2,
  hyperParameters: {
    value: '{"parameter_id": 2, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 128, "lr": 0.0001, "momentum": 0.4727135784273845}, "parameter_index": 0}',
    index: 0
  },
  placementConstraint: { type: 'None', gpus: [] }
}
[2022-01-20 04:48:04] INFO (NNIManager) Trial job Cwqkr status changed from WAITING to RUNNING
[2022-01-20 04:50:04] INFO (NNIManager) Trial job Cwqkr status changed from RUNNING to SUCCEEDED
[2022-01-20 04:50:05] INFO (NNIManager) NNIManager received command from dispatcher: TR, {"parameter_id": 3, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 512, "lr": 0.1, "momentum": 0.5413591322473582}, "parameter_index": 0}
[2022-01-20 04:50:10] INFO (NNIManager) submitTrialJob: form: {
  sequenceId: 3,
  hyperParameters: {
    value: '{"parameter_id": 3, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 512, "lr": 0.1, "momentum": 0.5413591322473582}, "parameter_index": 0}',
    index: 0
  },
  placementConstraint: { type: 'None', gpus: [] }
}
[2022-01-20 04:50:15] INFO (NNIManager) Trial job Nfbz9 status changed from WAITING to RUNNING
[2022-01-20 04:52:15] INFO (NNIManager) Trial job Nfbz9 status changed from RUNNING to SUCCEEDED
[2022-01-20 04:52:16] INFO (NNIManager) NNIManager received command from dispatcher: TR, {"parameter_id": 4, "parameter_source": "algorithm", "parameters": {"batch_size": 128, "hidden_size": 256, "lr": 0.0001, "momentum": 0.21466705347551573}, "parameter_index": 0}
[2022-01-20 04:52:21] INFO (NNIManager) submitTrialJob: form: {
  sequenceId: 4,
  hyperParameters: {
    value: '{"parameter_id": 4, "parameter_source": "algorithm", "parameters": {"batch_size": 128, "hidden_size": 256, "lr": 0.0001, "momentum": 0.21466705347551573}, "parameter_index": 0}',
    index: 0
  },
  placementConstraint: { type: 'None', gpus: [] }
}
[2022-01-20 04:52:26] INFO (NNIManager) Trial job lwoDF status changed from WAITING to RUNNING
[2022-01-20 04:54:16] INFO (NNIManager) Trial job lwoDF status changed from RUNNING to SUCCEEDED
[2022-01-20 04:54:17] INFO (NNIManager) NNIManager received command from dispatcher: TR, {"parameter_id": 5, "parameter_source": "algorithm", "parameters": {"batch_size": 32, "hidden_size": 128, "lr": 0.1, "momentum": 0.5981442137999028}, "parameter_index": 0}
[2022-01-20 04:54:22] INFO (NNIManager) submitTrialJob: form: {
  sequenceId: 5,
  hyperParameters: {
    value: '{"parameter_id": 5, "parameter_source": "algorithm", "parameters": {"batch_size": 32, "hidden_size": 128, "lr": 0.1, "momentum": 0.5981442137999028}, "parameter_index": 0}',
    index: 0
  },
  placementConstraint: { type: 'None', gpus: [] }
}
[2022-01-20 04:54:27] INFO (NNIManager) Trial job Ps2UV status changed from WAITING to RUNNING
[2022-01-20 04:56:47] INFO (NNIManager) Trial job Ps2UV status changed from RUNNING to SUCCEEDED
[2022-01-20 04:56:48] INFO (NNIManager) NNIManager received command from dispatcher: TR, {"parameter_id": 6, "parameter_source": "algorithm", "parameters": {"batch_size": 128, "hidden_size": 128, "lr": 0.001, "momentum": 0.9064910421608436}, "parameter_index": 0}
[2022-01-20 04:56:53] INFO (NNIManager) submitTrialJob: form: {
  sequenceId: 6,
  hyperParameters: {
    value: '{"parameter_id": 6, "parameter_source": "algorithm", "parameters": {"batch_size": 128, "hidden_size": 128, "lr": 0.001, "momentum": 0.9064910421608436}, "parameter_index": 0}',
    index: 0
  },
  placementConstraint: { type: 'None', gpus: [] }
}

 - dispatcher.log:
 [2022-01-20 04:43:24] INFO (nni.experiment/MainThread) Creating experiment, Experiment ID: [36m0zv4dxuc[0m
[2022-01-20 04:43:24] INFO (nni.experiment/MainThread) Starting web server...
[2022-01-20 04:43:25] WARNING (nni.experiment/MainThread) Timeout, retry...
[2022-01-20 04:43:26] INFO (nni.experiment/MainThread) Setting up...
[2022-01-20 04:43:26] INFO (nni.experiment/MainThread) Web UI URLs: [36mhttp://127.0.0.1:8080 http://192.168.55.106:8080[0m
[2022-01-20 04:43:26] INFO (nni.tools.nnictl.launcher/MainThread) To stop experiment run "nnictl stop 0zv4dxuc" or "nnictl stop --all"
[2022-01-20 04:43:26] INFO (nni.tools.nnictl.launcher/MainThread) Reference: https://nni.readthedocs.io/en/stable/Tutorial/Nnictl.html
[2022-01-20 04:43:27] INFO (nni.tuner.tpe/MainThread) Using random seed 755466869
[2022-01-20 04:43:27] INFO (nni.runtime.msg_dispatcher_base/MainThread) Dispatcher started

- trial.log
[2022-01-20 04:43:37] PRINT {'data_dir': './data', 'batch_size': 64, 'batch_num': None, 'hidden_size': 512, 'lr': 0.0001, 'momentum': 0.7760213206833312, 'epochs': 10, 'seed': 1, 'no_cuda': False, 'log_interval': 1000}
[2022-01-20 04:43:40] INFO (mnist_AutoML/MainThread) Train Epoch: 1 [0/60000 (0%)]	Loss: 2.312253
[2022-01-20 04:43:51] INFO (mnist_AutoML/MainThread) 
Test set: Average loss: 2.1372, Accuracy: 5719/10000 (57%)

[2022-01-20 04:43:51] INFO (mnist_AutoML/MainThread) Train Epoch: 2 [0/60000 (0%)]	Loss: 2.144817
[2022-01-20 04:44:02] INFO (mnist_AutoML/MainThread) 
Test set: Average loss: 1.6174, Accuracy: 7143/10000 (71%)

[2022-01-20 04:44:02] INFO (mnist_AutoML/MainThread) Train Epoch: 3 [0/60000 (0%)]	Loss: 1.604976
[2022-01-20 04:44:14] INFO (mnist_AutoML/MainThread) 
Test set: Average loss: 0.8386, Accuracy: 8141/10000 (81%)

[2022-01-20 04:44:14] INFO (mnist_AutoML/MainThread) Train Epoch: 4 [0/60000 (0%)]	Loss: 1.001007
[2022-01-20 04:44:25] INFO (mnist_AutoML/MainThread) 
Test set: Average loss: 0.5419, Accuracy: 8583/10000 (86%)

[2022-01-20 04:44:25] INFO (mnist_AutoML/MainThread) Train Epoch: 5 [0/60000 (0%)]	Loss: 0.496065
[2022-01-20 04:44:37] INFO (mnist_AutoML/MainThread) 
Test set: Average loss: 0.4267, Accuracy: 8808/10000 (88%)

[2022-01-20 04:44:37] INFO (mnist_AutoML/MainThread) Train Epoch: 6 [0/60000 (0%)]	Loss: 0.400071
[2022-01-20 04:44:49] INFO (mnist_AutoML/MainThread) 
Test set: Average loss: 0.3682, Accuracy: 8958/10000 (90%)

[2022-01-20 04:44:49] INFO (mnist_AutoML/MainThread) Train Epoch: 7 [0/60000 (0%)]	Loss: 0.185010
[2022-01-20 04:45:01] INFO (mnist_AutoML/MainThread) 
Test set: Average loss: 0.3286, Accuracy: 9064/10000 (91%)

[2022-01-20 04:45:01] INFO (mnist_AutoML/MainThread) Train Epoch: 8 [0/60000 (0%)]	Loss: 0.295494
[2022-01-20 04:45:13] INFO (mnist_AutoML/MainThread) 
Test set: Average loss: 0.3036, Accuracy: 9119/10000 (91%)

[2022-01-20 04:45:13] INFO (mnist_AutoML/MainThread) Train Epoch: 9 [0/60000 (0%)]	Loss: 0.284811
[2022-01-20 04:45:25] INFO (mnist_AutoML/MainThread) 
Test set: Average loss: 0.2797, Accuracy: 9184/10000 (92%)

[2022-01-20 04:45:25] INFO (mnist_AutoML/MainThread) Train Epoch: 10 [0/60000 (0%)]	Loss: 0.246093
[2022-01-20 04:45:37] INFO (mnist_AutoML/MainThread) 
Test set: Average loss: 0.2610, Accuracy: 9239/10000 (92%)

- no stdout no stderr 

**How to reproduce it?**:
just copy my config.yml and paste into yours, then run nnictl create --config nni/examples/trials/mnist-pytorch/config.yml command.

**Comment by liuzhe-lz** [comment]
You can add `logging.info('CUDA available: %s', torch.cuda.is_available())` to your trial code, and check trial's log file.

**Comment by RuojiWang** [comment]
> You can add `logging.info('CUDA available: %s', torch.cuda.is_available())` to your trial code, and check trial's log file.

thanks, now i can see the logo information, and i am sure CUDA is working.

but i have anthor question: how to set the suitable epoch_num of my experiment? i mean for different datasets the best epoch_num are different, can i use nni search that for me in condition that i do not know the search space of epoch_num(so i will not set epoch_num in my seach_space.json)

**Comment by scarlett2018** [comment]
creating a new issue for the new question and closing this resolved question.
